en/services/parameter_ext.md

```diff
@@ -0,0 +1,238 @@
+<span id="PARAM_EXT_REQUEST_READ"></span>[PARAM_EXT_REQUEST_READ](../messages/common.md#PARAM_EXT_REQUEST_READ) | Request the value of a specific parameter using either its `param_id` or `param_index`. Expects a response in a [PARAM_EXT_VALUE](#PARAM_EXT_VALUE).
+<span id="PARAM_EXT_SET"></span>[PARAM_EXT_SET](../messages/common.md#PARAM_EXT_SET) | Set a parameter value. Expects immediate response [PARAM_EXT_ACK](#PARAM_EXT_ACK) with result indicating success, failure, or that the request is still in progress (`PARAM_ACK_IN_PROGRESS`). If in progress, additional update [PARAM_EXT_ACK](#PARAM_EXT_ACK) messages are expected.
+<span id="PARAM_EXT_ACK"></span>[PARAM_EXT_ACK](../messages/common.md#PARAM_EXT_ACK) | Response from a [PARAM_EXT_SET](#PARAM_EXT_SET) message, which indicates whether the value was accepted (set), failed, setting is still in progress, or that the specified parameter is invalid/unsupported.
+
+Enum | Description
+-- | --
+<span id="MAV_PARAM_EXT_TYPE"></span>[MAV_PARAM_EXT_TYPE](../messages/common.md#MAV_PARAM_EXT_TYPE) | Specifies the datatype of a MAVLink extended parameter (parameter values are [encoded](parameter_encoding) within the a `char[128]` array in the messages). This type conveys the *real type* of the encoded parameter value, e.g. `MAV_PARAM_EXT_TYPE_REAL32`.
+<span id="PARAM_ACK"></span>[PARAM_ACK](../messages/common.md#PARAM_ACK) | Request acknowledgment status value, sent in an [PARAM_EXT_ACK](#PARAM_EXT_ACK) as a response to a [PARAM_EXT_SET](#PARAM_EXT_SET) message. A request can be accepted, fail, in-progress, or unsupported (indicating the specified parameter does not exist or has an invalid value or value type).
+
+
+## Parameter Encoding {#parameter_encoding}
+
+Parameters names/ids are set in the `param_id` field of messages where they are used.
+The `param_id` string can store up to 16 characters. 
+The string is terminated with a NULL (`\0`) character if there are less than 16 human-readable chars, and without a null termination byte if the length is exactly 16 chars.
+
+> **Note** Names (as above) are the same as for the [Parameter Protocol](../services/parameter.md#parameter_encoding).
+  Values (below) are encoded differently.
+
+Values are byte-wise encoded *within* the `param_value` field, which is a `char[128]`. <!-- how are they encoded? -->
+The `param_type` ([MAV_PARAM_EXT_TYPE](#MAV_PARAM_EXT_TYPE)) is used to indicate the actual type of the data so that it can be decoded by the recipient.
+Supported types are: 8, 16, 32 and 64-bit signed and unsigned integers, 32 and 64-bit floating point numbers, and a "custom type" which may used for strings. <!-- why custom? shouldit be string, and then new types added for non string usage of the same field -->
```

Comment:
The last type is a "Custom Type". How come we didn't have an explicit string type, allowing further random specialization via a custom type?

Reply:
The contents of the `char[128]` field is "memcpy" into a union structure:
https://github.com/mavlink/qgroundcontrol/blob/master/src/Camera/QGCCameraIO.h#L18-L34

Here is an example of code that _sends a parameter_ (Sets a value into the `char[128]` field):
https://github.com/mavlink/qgroundcontrol/blob/master/src/Camera/QGCCameraIO.cc#L132

Here is the other end, the code that receives a `char[128]` and takes the appropriate (typed) value off of it:
https://github.com/mavlink/qgroundcontrol/blob/master/src/Camera/QGCCameraIO.cc#L276